core/src/main/scala/org/apache/spark/internal/config/History.scala

```diff
@@ -125,4 +125,18 @@ private[spark] object History {
   val KERBEROS_KEYTAB = ConfigBuilder("spark.history.kerberos.keytab")
     .stringConf
     .createOptional
+
+  val CUSTOM_EXECUTOR_LOG_URL = ConfigBuilder("spark.history.custom.executor.log.url")
+    .doc("Specifies custom spark executor log url for supporting external log service instead of " +
+      "using resource managers' app log urls. Spark will support some path variables via " +
+      "patterns which can be vary on resource managers. Please check the document for " +
+      "resource manager to refer which patterns are supported.")
+    .stringConf
+    .createOptional
+
+  val APPLY_CUSTOM_EXECUTOR_LOG_URL_TO_INCOMPLETE_APP =
+    ConfigBuilder("spark.history.custom.executor.log.url.applyIncompleteApplication")
+      .doc("Whether apply custom executor log url to incomplete application as well")
```

Comment:
Whether apply custom executor log url, as specified by "spark.history.custom.executor.log.url", to incomplete application as well.  Even if this is true, this still only affects the behavior of the history server, not running spark applications.